packages/sdk/src/services/client-task-queue-service.ts

```diff
@@ -64,6 +83,99 @@ export class ClientTaskQueueService {
       in_progress: inProgressTasks.client_tasks.length,
     };
   }
+
+  public async executeClientTasks(
+    listOfClientTasks: ClientTask[],
+    authData: AuthData
+  ): Promise<{ completedTasks: ClientTask[]; failedTasks: ClientTask[] }> {
+    const remainingClientTasks: ClientTask[] = [];
+    const itemUpdateSharesTasks: ClientTask[] = [];
+    for (const task of listOfClientTasks) {
+      switch (task.work_type) {
+        case 'update_item_shares':
+          itemUpdateSharesTasks.push(task);
+          break;
+        default:
+          remainingClientTasks.push(task);
+          break;
+      }
+    }
+    if (remainingClientTasks.length) {
+      throw new Error(
+        `Do not know how to execute ClientTask of type ${remainingClientTasks[0].work_type}`
+      );
+    }
+    const [updateSharesTasksResult]: Array<{
+      completedTasks: ClientTask[];
+      failedTasks: IFailedClientTask[];
+    }> = await Promise.all([this.updateSharesClientTasks(itemUpdateSharesTasks, authData)]);
+
+    return updateSharesTasksResult;
+  }
+
+  public async updateSharesClientTasks(
+    listOfClientTasks: ClientTask[],
+    authData: AuthData
+  ): Promise<{ completedTasks: ClientTask[]; failedTasks: IFailedClientTask[] }> {
+    const sharesApi = this.vaultAPIFactory(authData.vault_access_token).SharesApi;
+    const itemsApi = this.vaultAPIFactory(authData.vault_access_token).ItemApi;
+
+    const taskReports = await Promise.all(
+      listOfClientTasks.map(async task => {
+        const taskReport = {
+          completedTasks: <ClientTask[]>[],
+          failedTasks: <IFailedClientTask[]>[],
+        };
+        try {
+          const [item, shares] = await Promise.all([
+            itemsApi.itemsIdGet(task.target_id),
+            sharesApi.itemsIdSharesGet(task.target_id),
+          ]);
+          const decryptedSlots = await ItemService.decryptAllSlots(
+            item.slots,
+            authData.data_encryption_key
+          );
+          const dek = this.cryppo.generateRandomKey();
```

Comment:
`generateRandomKey` Returns a promise I'm pretty sure?

Reply:
@zbarbuto surprisingly not 
it is just random bytes
`export const generateRandomKey = (length = 32) => random.getBytesSync(length);`